app/src/main/java/org/simple/clinic/newentry/PatientEntryLogic.kt

```diff
@@ -15,24 +15,24 @@ fun patientEntryUpdate(
     model: PatientEntryModel,
     event: PatientEntryEvent
 ): Next<PatientEntryModel, PatientEntryEffect> {
-  if (event is OngoingEntryFetched) {
-    return next(
+  return when (event) {
+    is OngoingEntryFetched -> next(
         model.patientEntryFetched(event.patientEntry),
         setOf(PrefillFields(event.patientEntry))
     )
-  } else if (event is GenderChanged) {
-    val updatedModel = model.withGender(event.gender)
 
-    if (event.gender.isNotEmpty() && model.isSelectingGenderForTheFirstTime) {
-      return next(updatedModel.copy(isSelectingGenderForTheFirstTime = false), setOf(ScrollFormToBottom))
-    } else {
-      return next(updatedModel)
+    is GenderChanged -> {
+      val updatedModel = model.withGender(event.gender)
+
+      return if (event.gender.isNotEmpty() && model.isSelectingGenderForTheFirstTime) {
```

Comment:
This `return` is redundant since the `when` block is already being returned.

Reply:
I have moved all the logic out of the controller and have begun refactoring this section. Can we make it part of the next PR?